### PR TITLE
Add a tool to check cross references

### PR DESCRIPTION
**Pull request by gsmet**
@yrodiere this is a follow up of our discussion.

There's no easy way to collect all the ids so it is done on a best effort basis. All the cases in the current documentation are covered.

Unfortunately, it doesn't support additional ids added just in the
title as in:
```
[id1]
== [id2] [id3] Title
```
so it will force people to use the canonical id but I think it's
acceptable in our reference doc to enforce the canonical id.

Some other things are a bit ad hoc but it somehow works and I'll maintain it to keep it working.

Obviously, I fixed all the issues it detected.

Also move the link transformation to downstream documentation only as
this alters the ability of Asciidoctor to check the internal links.

### CODE REVIEW

File: docs/src/main/asciidoc/cdi-reference.adoc [hunk 1]
```diff
@@ -1072,9 +1072,10 @@ The downside of this approach is that CDI Portable Extensions cannot be supporte
 Nevertheless, most of the functionality can be achieved using Quarkus xref:writing-extensions.adoc[extensions].
 See the xref:cdi-integration.adoc[integration guide] for more information.
 
-== Development Mode
+[[dev_mode]]
+== Dev mode
```
Comment (yrodiere) [inline docs/src/main/asciidoc/cdi-reference.adoc:1076]: This kind of change breaks existing links, please let's not do that :(

```suggestion
[[dev_mode]]
== [[development-mode]] Dev mode
```

File: docs/src/main/asciidoc/redis-reference.adoc [hunk 2]
```diff
@@ -87,6 +87,7 @@ To help you select the suitable API for you, here are some recommendations:
 * If you have existing Vert.x code, use `io.vertx.redis.client.RedisAPI`
 * If you need to emit custom commands, you can either use the data sources (reactive or imperative) or the `io.vertx.mutiny.redis.client.Redis`.
 
+[[default-and-named-clients]]
 == Inject the default and named clients
```
Comment (yrodiere) [inline docs/src/main/asciidoc/redis-reference.adoc:91]:
```suggestion
[[default-and-named-clients]]
== [[inject-the-default-and-named-clients]] Inject the default and named clients
```

File: docs/src/main/asciidoc/resteasy-reactive.adoc [hunk 3]
```diff
@@ -394,9 +394,8 @@ public class Endpoint {
 ----
 <1> `BeanParam` is required to comply with the Jakarta REST specification so that libraries like OpenAPI can introspect the parameters.
 
-=== Declaring URI parameters
-
 [[uri-parameters]]
+=== Declaring URI parameters
```
Comment (yrodiere) [inline docs/src/main/asciidoc/resteasy-reactive.adoc:398]:
```suggestion
[[uri-parameters]]
=== [[declaring-uri-parameters]] Declaring URI parameters
```

**Review by yrodiere — APPROVED** [review]
> There's no easy way to collect all the ids so it is done on a best effort basis. All the cases in the current documentation are covered.

Yeah unless we integrate into an actual asciidoc renderer, I don't see how we could do better than "best-effort". Sounds good.

> so it will force people to use the canonical id but I think it's
> acceptable in our reference doc to enforce the canonical id.

+1 sounds fine. Additional IDs are only there to keep others links (from other websites) working anyway.

File: docs/src/main/asciidoc/cdi-reference.adoc (hunk 1 above)
Comment (gsmet) [reply docs/src/main/asciidoc/cdi-reference.adoc:1076]: We could do that but I'm pretty sure nobody will do it when adding an explicit anchor in the future so...

Comment (yrodiere) [reply docs/src/main/asciidoc/cdi-reference.adoc]: I'm pretty sure "others don't do it" has never stopped you from doing things right :)

**Review by MichalMaler — APPROVED** [review]

**Pull request merged**